src/Microsoft.DotNet.Cli.Utils/Telemetry.cs

```diff
@@ -13,6 +13,7 @@ public class Telemetry
         private static Dictionary<string, string> _commonProperties = null;
         private static Dictionary<string, double> _commonMeasurements = null;
 
+        //readonly instead of const to to avoid inlining in case we need to change the instrumentation key
```

Comment:
I don't think this makes sense. The field is private, which means only this class can access it. If you change the instrumentation key, this class will need to be recompiled. Which means anything that can access the field will need to be recompiled.
Thus you don't need to worry about inlining or not. Any time this value changes, all of its consumers need to be recompiled anyway.